mappings/net/minecraft/entity/Mountable.mapping

```diff
@@ -1,4 +1,5 @@
-CLASS net/minecraft/class_5146
+CLASS net/minecraft/class_5146 net/minecraft/entity/Mountable
```

Comment:
More like saddleable, see #1269

Reply:
Entities with that interface are a Mount, which makes them _mountable_. I find this name more grammatical.

Reply:
Yeah, Saddleable is incorrect because not all entities with it can be "saddled" which sounds wrong anyway. `Mountable` or even just `Mount` is a more accurate name.

Reply:
No, this is just saddle logic and is mainly used by dispenser to install saddles. The method `setSaddled` is just installing saddle and passing the sound category for the saddling sound to play in (i.e. block sound category for dispensers and player sound category for players)

the other two methods checks if the entity already has saddle or if it can have saddle installed at current state

Reply:
If canBeSaddled is false, then that means the entity is not saddleable and therefore the method name is incorrect. See how llamas always return false, an entity that is *not* saddleable and implements the interface.

Reply:
Yes because one of its parent classes is saddleable. Show me how this interface ever relates to mounting logic. Also this class is already mapped in latest yarn

Reply:
It's an interface that every mountable animal in the game implements, and the ones that can be saddleable return true on canBeSaddled. It doesn't get much simpler than that.

Reply:
![image](https://user-images.githubusercontent.com/7806504/78868022-c19ef000-7a07-11ea-92f2-67420b7e028e.png)
As far as I see from decompiled code this interface is 100% saddle related. It has nothing to do with "mounting".

@Prospector please read the code. How about minecarts boats, which are also mountable? If your theory is right, how about ravagers, which you consider mountable, but cannot have saddles installed (because they always come with one)?

Reply:
You keep dodging the fact that an interface you named `Saddleable` is used by entities which factually are not saddleable. Therefore, it's an incorrect name. The common link between the things that implement it are not whether or not they are saddleable, it's whether or not they are mountable. It's like naming `BlockWithEntity` something like `ScreenHandlerEntityBlock`. While yes, the interface does enable screen handler functionality within some subclasses, it is still implemented by blocks which do not use screen handlers, and therefore cannot be named after that part of its functionality.

You did bring up a good point, and that is that there are mobs that are mountable by other entities such as ravagers, chickens, and spiders. Therefore, I'd propose the name `PlayerMountable`.

Reply:
> You keep dodging the fact that an interface you named Saddleable is used by entities which factually are not saddleable.

Because mojang doesn't want to refactor llama out of the horse inheritance tree just because of that saddleable interface. End of the story. Nowhere in code is this ever related to player mounting at all. If you name it, show evidence in code.

Again, again, again, we need _**evidence**_.

> The common link between the things that implement it are not whether or not they are saddleable, it's whether or not they are mountable.

There is no mounting logic touched in this interface at all. It is just a coincidence that most tameable + mountable entities need saddles to mount, and the only one that does not need saddles extends from one that need saddles (Llama extends Donkey).

Your logic of basing this off this coincidence is just like [attributing the correlation between ice cream sale and crime rate as causation](https://slate.com/news-and-politics/2013/07/warm-weather-homicide-rates-when-ice-cream-sales-rise-homicides-rise-coincidence.html).

Reply:
> How about minecarts boats, which are also mountable?

You don't "mount" a minecart or boat, you "board" them. However you can "ride" both horses and minecarts.